### PR TITLE
4 get api articles articleid

### PR DESCRIPTION
**Pull request by Gazdean**
ticket 4

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -63,4 +63,52 @@ describe('error handling for all invalid paths', () => {
...
+
+            const {article} = body
+            expect(article).toHaveProperty('author')
+            expect(article).toHaveProperty('article_id')
+            expect(article).toHaveProperty('title')
+            expect(article).toHaveProperty('body')
+            expect(article).toHaveProperty('topic')
+            expect(article).toHaveProperty('created_at')
+            expect(article).toHaveProperty('votes')
+            expect(article).toHaveProperty('article_img_url')
```
Comment (NRMSMITH) [inline __tests__/api.test.js]: I would suggest using .toMatchObject or .objectContaining here to make this more flexible for later on in your test suite :)

File: db/app.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+        .send({ mess: err.mess });
+    }
+    next(err);
+})
+app.use((err, req, res, next) => {
+    if (err.code === '22P02') {
+      res.status(400).send({mess: 'bad request'});
+    } else next(err)
+  });
+
```
Comment (NRMSMITH) [inline db/app.js]: Your Error handlers are becoming quite big here - I think now is the time to pop them into a errors.controllers.js :)

File: db/controllers/articles-controller.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const {fetchArticlesById} = require('../models/articles-model')
+
+exports.getArticlesById = (req, res, next) => {
+    const { article_id } = req.params;
+    fetchArticlesById(article_id)
+        .then(({article}) => {
+        res.status(200).send({ article });
+    })
+    .catch(next)
+};
```
Comment (NRMSMITH) [inline db/controllers/articles-controller.js:10]: Lovely and neat

File: playground.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+You are now connected to database "nc_news_test" as user "garethdean".
+ article_id | title | topic | author | body | created_at | votes | article_img_url 
+------------+-------+-------+--------+------+------------+-------+-----------------
+(0 rows)
+
```
Comment (NRMSMITH) [inline playground.txt:5]: pop this in your .gitignore please :)

**Review by NRMSMITH — APPROVED** [review]
I've made a few suggestions - feel free to merge once they're done :)

File: playground.txt (hunk 4 above)
Comment (Gazdean) [reply playground.txt:5]: oh yes, apologies
done

File: db/app.js (hunk 2 above)
Comment (Gazdean) [reply db/app.js]: done

File: db/controllers/articles-controller.js (hunk 3 above)
Comment (Gazdean) [reply db/controllers/articles-controller.js:10]: thankyou

File: __tests__/api.test.js (hunk 1 above)
Comment (Gazdean) [reply __tests__/api.test.js]: used .toMatchObject

**Comment by Gazdean** [comment]
merged and deleted branch

**Pull request merged**